File: backend/src/main/java/com/godeltech/mastery/backend/domain/dto/responce/CarDTO.java
package com.godeltech.mastery.backend.domain.dto.responce;

import java.math.BigDecimal;
import java.util.List;

public record CarDTO(
  Long id,
  BrandDTO brand,
  ModelDTO model,
  Integer year,
  String photoUrl,
  String vin,
  Integer mileage,
  Boolean ours,
  ClientDTO client,
  EquipmentDTO equipment,
  Boolean used,
  BigDecimal price,
  GuaranteeDTO guarantee,
  List<ServiceOperationRecordDTO> serviceOperations
) {}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/domain/dto/responce/CarDto.kt
package com.godeltech.mastery.backend.domain.dto.responce

data class CarDto(
    val id: Long?,
    val brand: BrandDto?,
    val model: ModelDto?,
    val year: Int?,
    val photoUrl: String?,
    val vin: String?,
    val mileage: Int?,
    val ours: Boolean?,
    val client: ClientDto?,
    val equipment: EquipmentDto?,
    val used: Boolean?,
    val price: Double?,
    val guarantee: GuaranteeDto,
    val serviceOperations: List<ServiceOperationRecordDto>?
)
